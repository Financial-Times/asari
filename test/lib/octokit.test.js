const nock = require('nock');
const authenticatedOctokit = require("../../lib/octokit")

// Don't let Octokit make network requests
nock.disableNetConnect();

describe("lib/octokit.js", () => {
	test("Calling octokit() without a personalAccessToken throws an error", async () => {
		expect.assertions(1)
		try {
			await octokit({})
		}
		catch (error) {
			expect(error).toBeInstanceOf(Error)
		}
	})

	test("Calling octokit() returns an instance of Octokit", async () => {
		const octokit = await authenticatedOctokit({ personalAccessToken: "incorrectToken" })
		expect(octokit).toEqual(
			expect.objectContaining({
				git: expect.any(Object),
				authenticate: expect.any(Function),
			})
		)
	})

	test("Calling octokit() triggers a POST network request to https://api.github.com", async () => {

		// If this endpoint is not called, nock.isDone() will be false.
		nock('https://api.github.com')
			.persist()
			.post('/repos/test/test/pulls')
			.reply(200, {})

		const octokit = await authenticatedOctokit({ personalAccessToken: "incorrectToken" })
		await octokit.pulls.create({
			owner: "test",
			repo: "test",
			title: "test",
			head: "test",
			base: "test",
			body: "test",
		})
		expect(nock.isDone()).toBe(true)
	})
})
