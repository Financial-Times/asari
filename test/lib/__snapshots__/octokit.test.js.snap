// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/octokit.js Calling octokit() returns an instance of Octokit 1`] = `
Object {
  "activity": Object {
    "checkStarringRepo": [Function],
    "deleteRepoSubscription": [Function],
    "deleteThreadSubscription": [Function],
    "getRepoSubscription": [Function],
    "getThread": [Function],
    "getThreadSubscription": [Function],
    "listEventsForOrg": [Function],
    "listEventsForUser": [Function],
    "listFeeds": [Function],
    "listNotifications": [Function],
    "listNotificationsForRepo": [Function],
    "listPublicEvents": [Function],
    "listPublicEventsForOrg": [Function],
    "listPublicEventsForRepoNetwork": [Function],
    "listPublicEventsForUser": [Function],
    "listReceivedEventsForUser": [Function],
    "listReceivedPublicEventsForUser": [Function],
    "listRepoEvents": [Function],
    "listReposStarredByAuthenticatedUser": [Function],
    "listReposStarredByUser": [Function],
    "listReposWatchedByUser": [Function],
    "listStargazersForRepo": [Function],
    "listWatchedReposForAuthenticatedUser": [Function],
    "listWatchersForRepo": [Function],
    "markAsRead": [Function],
    "markNotificationsAsReadForRepo": [Function],
    "markThreadAsRead": [Function],
    "setRepoSubscription": [Function],
    "setThreadSubscription": [Function],
    "starRepo": [Function],
    "unstarRepo": [Function],
  },
  "apps": Object {
    "addRepoToInstallation": [Function],
    "checkAccountIsAssociatedWithAny": [Function],
    "checkAccountIsAssociatedWithAnyStubbed": [Function],
    "createContentAttachment": [Function],
    "createFromManifest": [Function],
    "createInstallationToken": [Function],
    "findOrgInstallation": [Function],
    "findRepoInstallation": [Function],
    "findUserInstallation": [Function],
    "getAuthenticated": [Function],
    "getBySlug": [Function],
    "getInstallation": [Function],
    "listAccountsUserOrOrgOnPlan": [Function],
    "listAccountsUserOrOrgOnPlanStubbed": [Function],
    "listInstallationReposForAuthenticatedUser": [Function],
    "listInstallations": [Function],
    "listInstallationsForAuthenticatedUser": [Function],
    "listMarketplacePurchasesForAuthenticatedUser": [Function],
    "listMarketplacePurchasesForAuthenticatedUserStubbed": [Function],
    "listPlans": [Function],
    "listPlansStubbed": [Function],
    "listRepos": [Function],
    "removeRepoFromInstallation": [Function],
  },
  "authenticate": [Function],
  "authorization": Object {
    "checkAuthorization": [Function],
    "createAuthorization": [Function],
    "deleteAuthorization": [Function],
    "deleteGrant": [Function],
    "getAuthorization": [Function],
    "getGrant": [Function],
    "getOrCreateAuthorizationForApp": [Function],
    "getOrCreateAuthorizationForAppAndFingerprint": [Function],
    "getOrCreateAuthorizationForAppFingerprint": [Function],
    "listAuthorizations": [Function],
    "listGrants": [Function],
    "resetAuthorization": [Function],
    "revokeAuthorizationForApplication": [Function],
    "revokeGrantForApplication": [Function],
    "updateAuthorization": [Function],
  },
  "checks": Object {
    "create": [Function],
    "createSuite": [Function],
    "get": [Function],
    "getSuite": [Function],
    "listAnnotations": [Function],
    "listForRef": [Function],
    "listForSuite": [Function],
    "listSuitesForRef": [Function],
    "rerequestSuite": [Function],
    "setSuitesPreferences": [Function],
    "update": [Function],
  },
  "codesOfConduct": Object {
    "getConductCode": [Function],
    "getForRepo": [Function],
    "listConductCodes": [Function],
  },
  "emojis": Object {
    "get": [Function],
  },
  "getFirstPage": [Function],
  "getLastPage": [Function],
  "getNextPage": [Function],
  "getPreviousPage": [Function],
  "gists": Object {
    "checkIsStarred": [Function],
    "create": [Function],
    "createComment": [Function],
    "delete": [Function],
    "deleteComment": [Function],
    "fork": [Function],
    "get": [Function],
    "getComment": [Function],
    "getRevision": [Function],
    "list": [Function],
    "listComments": [Function],
    "listCommits": [Function],
    "listForks": [Function],
    "listPublic": [Function],
    "listPublicForUser": [Function],
    "listStarred": [Function],
    "star": [Function],
    "unstar": [Function],
    "update": [Function],
    "updateComment": [Function],
  },
  "git": Object {
    "createBlob": [Function],
    "createCommit": [Function],
    "createRef": [Function],
    "createTag": [Function],
    "createTree": [Function],
    "deleteRef": [Function],
    "getBlob": [Function],
    "getCommit": [Function],
    "getRef": [Function],
    "getTag": [Function],
    "getTree": [Function],
    "listRefs": [Function],
    "updateRef": [Function],
  },
  "gitdata": Object {
    "createBlob": [Function],
    "createCommit": [Function],
    "createRef": [Function],
    "createTag": [Function],
    "createTree": [Function],
    "deleteRef": [Function],
    "getBlob": [Function],
    "getCommit": [Function],
    "getRef": [Function],
    "getTag": [Function],
    "getTree": [Function],
    "listRefs": [Function],
    "updateRef": [Function],
  },
  "gitignore": Object {
    "getTemplate": [Function],
    "listTemplates": [Function],
  },
  "hasFirstPage": [Function],
  "hasLastPage": [Function],
  "hasNextPage": [Function],
  "hasPreviousPage": [Function],
  "hook": [Function],
  "interactions": Object {
    "addOrUpdateRestrictionsForOrg": [Function],
    "addOrUpdateRestrictionsForRepo": [Function],
    "getRestrictionsForOrg": [Function],
    "getRestrictionsForRepo": [Function],
    "removeRestrictionsForOrg": [Function],
    "removeRestrictionsForRepo": [Function],
  },
  "issues": Object {
    "addAssignees": [Function],
    "addLabels": [Function],
    "checkAssignee": [Function],
    "create": [Function],
    "createComment": [Function],
    "createLabel": [Function],
    "createMilestone": [Function],
    "deleteComment": [Function],
    "deleteLabel": [Function],
    "deleteMilestone": [Function],
    "get": [Function],
    "getComment": [Function],
    "getEvent": [Function],
    "getLabel": [Function],
    "getMilestone": [Function],
    "list": [Function],
    "listAssignees": [Function],
    "listComments": [Function],
    "listCommentsForRepo": [Function],
    "listEvents": [Function],
    "listEventsForRepo": [Function],
    "listEventsForTimeline": [Function],
    "listForAuthenticatedUser": [Function],
    "listForOrg": [Function],
    "listForRepo": [Function],
    "listLabelsForMilestone": [Function],
    "listLabelsForRepo": [Function],
    "listLabelsOnIssue": [Function],
    "listMilestonesForRepo": [Function],
    "lock": [Function],
    "removeAssignees": [Function],
    "removeLabel": [Function],
    "removeLabels": [Function],
    "replaceLabels": [Function],
    "unlock": [Function],
    "update": [Function],
    "updateComment": [Function],
    "updateLabel": [Function],
    "updateMilestone": [Function],
  },
  "licenses": Object {
    "get": [Function],
    "getForRepo": [Function],
    "list": [Function],
    "listCommonlyUsed": [Function],
  },
  "log": Object {
    "debug": [Function],
    "error": [Function],
    "info": [Function],
    "warn": [Function],
  },
  "markdown": Object {
    "render": [Function],
    "renderRaw": [Function],
  },
  "meta": Object {
    "get": [Function],
  },
  "migrations": Object {
    "cancelImport": [Function],
    "deleteArchiveForAuthenticatedUser": [Function],
    "deleteArchiveForOrg": [Function],
    "getArchiveForAuthenticatedUser": [Function],
    "getArchiveForOrg": [Function],
    "getCommitAuthors": [Function],
    "getImportProgress": [Function],
    "getLargeFiles": [Function],
    "getStatusForAuthenticatedUser": [Function],
    "getStatusForOrg": [Function],
    "listForAuthenticatedUser": [Function],
    "listForOrg": [Function],
    "mapCommitAuthor": [Function],
    "setLfsPreference": [Function],
    "startForAuthenticatedUser": [Function],
    "startForOrg": [Function],
    "startImport": [Function],
    "unlockRepoForAuthenticatedUser": [Function],
    "unlockRepoForOrg": [Function],
    "updateImport": [Function],
  },
  "oauthAuthorizations": Object {
    "checkAuthorization": [Function],
    "createAuthorization": [Function],
    "deleteAuthorization": [Function],
    "deleteGrant": [Function],
    "getAuthorization": [Function],
    "getGrant": [Function],
    "getOrCreateAuthorizationForApp": [Function],
    "getOrCreateAuthorizationForAppAndFingerprint": [Function],
    "getOrCreateAuthorizationForAppFingerprint": [Function],
    "listAuthorizations": [Function],
    "listGrants": [Function],
    "resetAuthorization": [Function],
    "revokeAuthorizationForApplication": [Function],
    "revokeGrantForApplication": [Function],
    "updateAuthorization": [Function],
  },
  "orgs": Object {
    "addOrUpdateMembership": [Function],
    "blockUser": [Function],
    "checkBlockedUser": [Function],
    "checkMembership": [Function],
    "checkPublicMembership": [Function],
    "concealMembership": [Function],
    "convertMemberToOutsideCollaborator": [Function],
    "createHook": [Function],
    "createInvitation": [Function],
    "deleteHook": [Function],
    "get": [Function],
    "getHook": [Function],
    "getMembership": [Function],
    "getMembershipForAuthenticatedUser": [Function],
    "list": [Function],
    "listBlockedUsers": [Function],
    "listForAuthenticatedUser": [Function],
    "listForUser": [Function],
    "listHooks": [Function],
    "listInvitationTeams": [Function],
    "listMembers": [Function],
    "listMemberships": [Function],
    "listOutsideCollaborators": [Function],
    "listPendingInvitations": [Function],
    "listPublicMembers": [Function],
    "pingHook": [Function],
    "publicizeMembership": [Function],
    "removeMember": [Function],
    "removeMembership": [Function],
    "removeOutsideCollaborator": [Function],
    "unblockUser": [Function],
    "update": [Function],
    "updateHook": [Function],
    "updateMembership": [Function],
  },
  "paginate": [Function],
  "projects": Object {
    "addCollaborator": [Function],
    "createCard": [Function],
    "createColumn": [Function],
    "createForAuthenticatedUser": [Function],
    "createForOrg": [Function],
    "createForRepo": [Function],
    "delete": [Function],
    "deleteCard": [Function],
    "deleteColumn": [Function],
    "get": [Function],
    "getCard": [Function],
    "getColumn": [Function],
    "listCards": [Function],
    "listCollaborators": [Function],
    "listColumns": [Function],
    "listForOrg": [Function],
    "listForRepo": [Function],
    "listForUser": [Function],
    "moveCard": [Function],
    "moveColumn": [Function],
    "removeCollaborator": [Function],
    "reviewUserPermissionLevel": [Function],
    "update": [Function],
    "updateCard": [Function],
    "updateColumn": [Function],
  },
  "pullRequests": Object {
    "checkIfMerged": [Function],
    "create": [Function],
    "createComment": [Function],
    "createCommentReply": [Function],
    "createFromIssue": [Function],
    "createReview": [Function],
    "createReviewRequest": [Function],
    "deleteComment": [Function],
    "deletePendingReview": [Function],
    "deleteReviewRequest": [Function],
    "dismissReview": [Function],
    "get": [Function],
    "getComment": [Function],
    "getCommentsForReview": [Function],
    "getReview": [Function],
    "list": [Function],
    "listComments": [Function],
    "listCommentsForRepo": [Function],
    "listCommits": [Function],
    "listFiles": [Function],
    "listReviewRequests": [Function],
    "listReviews": [Function],
    "merge": [Function],
    "submitReview": [Function],
    "update": [Function],
    "updateComment": [Function],
    "updateReview": [Function],
  },
  "pulls": Object {
    "checkIfMerged": [Function],
    "create": [Function],
    "createComment": [Function],
    "createCommentReply": [Function],
    "createFromIssue": [Function],
    "createReview": [Function],
    "createReviewRequest": [Function],
    "deleteComment": [Function],
    "deletePendingReview": [Function],
    "deleteReviewRequest": [Function],
    "dismissReview": [Function],
    "get": [Function],
    "getComment": [Function],
    "getCommentsForReview": [Function],
    "getReview": [Function],
    "list": [Function],
    "listComments": [Function],
    "listCommentsForRepo": [Function],
    "listCommits": [Function],
    "listFiles": [Function],
    "listReviewRequests": [Function],
    "listReviews": [Function],
    "merge": [Function],
    "submitReview": [Function],
    "update": [Function],
    "updateComment": [Function],
    "updateReview": [Function],
  },
  "rateLimit": Object {
    "get": [Function],
  },
  "reactions": Object {
    "createForCommitComment": [Function],
    "createForIssue": [Function],
    "createForIssueComment": [Function],
    "createForPullRequestReviewComment": [Function],
    "createForTeamDiscussion": [Function],
    "createForTeamDiscussionComment": [Function],
    "delete": [Function],
    "listForCommitComment": [Function],
    "listForIssue": [Function],
    "listForIssueComment": [Function],
    "listForPullRequestReviewComment": [Function],
    "listForTeamDiscussion": [Function],
    "listForTeamDiscussionComment": [Function],
  },
  "registerEndpoints": [Function],
  "repos": Object {
    "acceptInvitation": [Function],
    "addCollaborator": [Function],
    "addDeployKey": [Function],
    "addProtectedBranchAdminEnforcement": [Function],
    "addProtectedBranchRequiredSignatures": [Function],
    "addProtectedBranchRequiredStatusChecksContexts": [Function],
    "addProtectedBranchTeamRestrictions": [Function],
    "addProtectedBranchUserRestrictions": [Function],
    "checkCollaborator": [Function],
    "compareCommits": [Function],
    "createCommitComment": [Function],
    "createDeployment": [Function],
    "createDeploymentStatus": [Function],
    "createFile": [Function],
    "createForAuthenticatedUser": [Function],
    "createFork": [Function],
    "createHook": [Function],
    "createInOrg": [Function],
    "createRelease": [Function],
    "createStatus": [Function],
    "declineInvitation": [Function],
    "delete": [Function],
    "deleteCommitComment": [Function],
    "deleteDownload": [Function],
    "deleteFile": [Function],
    "deleteHook": [Function],
    "deleteInvitation": [Function],
    "deleteRelease": [Function],
    "deleteReleaseAsset": [Function],
    "disablePagesSite": [Function],
    "enablePagesSite": [Function],
    "get": [Function],
    "getArchiveLink": [Function],
    "getBranch": [Function],
    "getBranchProtection": [Function],
    "getClones": [Function],
    "getCodeFrequencyStats": [Function],
    "getCollaboratorPermissionLevel": [Function],
    "getCombinedStatusForRef": [Function],
    "getCommit": [Function],
    "getCommitActivityStats": [Function],
    "getCommitComment": [Function],
    "getCommitRefSha": [Function],
    "getContents": [Function],
    "getContributorsStats": [Function],
    "getDeployKey": [Function],
    "getDeployment": [Function],
    "getDeploymentStatus": [Function],
    "getDownload": [Function],
    "getHook": [Function],
    "getLatestPagesBuild": [Function],
    "getLatestRelease": [Function],
    "getPages": [Function],
    "getPagesBuild": [Function],
    "getParticipationStats": [Function],
    "getProtectedBranchAdminEnforcement": [Function],
    "getProtectedBranchPullRequestReviewEnforcement": [Function],
    "getProtectedBranchRequiredSignatures": [Function],
    "getProtectedBranchRequiredStatusChecks": [Function],
    "getProtectedBranchRestrictions": [Function],
    "getPunchCardStats": [Function],
    "getReadme": [Function],
    "getRelease": [Function],
    "getReleaseAsset": [Function],
    "getReleaseByTag": [Function],
    "getTopPaths": [Function],
    "getTopReferrers": [Function],
    "getViews": [Function],
    "list": [Function],
    "listAssetsForRelease": [Function],
    "listBranches": [Function],
    "listCollaborators": [Function],
    "listCommentsForCommit": [Function],
    "listCommitComments": [Function],
    "listCommits": [Function],
    "listContributors": [Function],
    "listDeployKeys": [Function],
    "listDeploymentStatuses": [Function],
    "listDeployments": [Function],
    "listDownloads": [Function],
    "listForOrg": [Function],
    "listForUser": [Function],
    "listForks": [Function],
    "listHooks": [Function],
    "listInvitations": [Function],
    "listInvitationsForAuthenticatedUser": [Function],
    "listLanguages": [Function],
    "listPagesBuilds": [Function],
    "listProtectedBranchRequiredStatusChecksContexts": [Function],
    "listProtectedBranchTeamRestrictions": [Function],
    "listProtectedBranchUserRestrictions": [Function],
    "listPublic": [Function],
    "listReleases": [Function],
    "listStatusesForRef": [Function],
    "listTags": [Function],
    "listTeams": [Function],
    "listTopics": [Function],
    "merge": [Function],
    "pingHook": [Function],
    "removeBranchProtection": [Function],
    "removeCollaborator": [Function],
    "removeDeployKey": [Function],
    "removeProtectedBranchAdminEnforcement": [Function],
    "removeProtectedBranchPullRequestReviewEnforcement": [Function],
    "removeProtectedBranchRequiredSignatures": [Function],
    "removeProtectedBranchRequiredStatusChecks": [Function],
    "removeProtectedBranchRequiredStatusChecksContexts": [Function],
    "removeProtectedBranchRestrictions": [Function],
    "removeProtectedBranchTeamRestrictions": [Function],
    "removeProtectedBranchUserRestrictions": [Function],
    "replaceProtectedBranchRequiredStatusChecksContexts": [Function],
    "replaceProtectedBranchTeamRestrictions": [Function],
    "replaceProtectedBranchUserRestrictions": [Function],
    "replaceTopics": [Function],
    "requestPageBuild": [Function],
    "retrieveCommunityProfileMetrics": [Function],
    "testPushHook": [Function],
    "transfer": [Function],
    "update": [Function],
    "updateBranchProtection": [Function],
    "updateCommitComment": [Function],
    "updateFile": [Function],
    "updateHook": [Function],
    "updateInformationAboutPagesSite": [Function],
    "updateInvitation": [Function],
    "updateProtectedBranchPullRequestReviewEnforcement": [Function],
    "updateProtectedBranchRequiredStatusChecks": [Function],
    "updateRelease": [Function],
    "updateReleaseAsset": [Function],
    "uploadReleaseAsset": [Function],
  },
  "request": [Function],
  "search": Object {
    "code": [Function],
    "commits": [Function],
    "issues": [Function],
    "issuesAndPullRequests": [Function],
    "labels": [Function],
    "repos": [Function],
    "topics": [Function],
    "users": [Function],
  },
  "teams": Object {
    "addMember": [Function],
    "addOrUpdateMembership": [Function],
    "addOrUpdateProject": [Function],
    "addOrUpdateRepo": [Function],
    "checkManagesRepo": [Function],
    "create": [Function],
    "createDiscussion": [Function],
    "createDiscussionComment": [Function],
    "delete": [Function],
    "deleteDiscussion": [Function],
    "deleteDiscussionComment": [Function],
    "get": [Function],
    "getDiscussion": [Function],
    "getDiscussionComment": [Function],
    "getMember": [Function],
    "getMembership": [Function],
    "list": [Function],
    "listChild": [Function],
    "listDiscussionComments": [Function],
    "listDiscussions": [Function],
    "listForAuthenticatedUser": [Function],
    "listMembers": [Function],
    "listPendingInvitations": [Function],
    "listProjects": [Function],
    "listRepos": [Function],
    "removeMember": [Function],
    "removeMembership": [Function],
    "removeProject": [Function],
    "removeRepo": [Function],
    "reviewProject": [Function],
    "update": [Function],
    "updateDiscussion": [Function],
    "updateDiscussionComment": [Function],
  },
  "users": Object {
    "addEmails": [Function],
    "block": [Function],
    "checkBlocked": [Function],
    "checkFollowing": [Function],
    "checkFollowingForUser": [Function],
    "createGpgKey": [Function],
    "createPublicKey": [Function],
    "deleteEmails": [Function],
    "deleteGpgKey": [Function],
    "deletePublicKey": [Function],
    "follow": [Function],
    "getAuthenticated": [Function],
    "getByUsername": [Function],
    "getContextForUser": [Function],
    "getGpgKey": [Function],
    "getPublicKey": [Function],
    "list": [Function],
    "listBlocked": [Function],
    "listEmails": [Function],
    "listFollowersForAuthenticatedUser": [Function],
    "listFollowersForUser": [Function],
    "listFollowingForAuthenticatedUser": [Function],
    "listFollowingForUser": [Function],
    "listGpgKeys": [Function],
    "listGpgKeysForUser": [Function],
    "listPublicEmails": [Function],
    "listPublicKeys": [Function],
    "listPublicKeysForUser": [Function],
    "togglePrimaryEmailVisibility": [Function],
    "unblock": [Function],
    "unfollow": [Function],
    "updateAuthenticated": [Function],
  },
}
`;
